### DISCUSSION TITLE
How to change amp URL in next.js?

### DISCUSSION THREAD

**Discussion by yudiz-kuldip** [vercel/next.js#37427] [Help] [answered]
I want to change the amp URL from this **?amp=1** to **?amp**.
If anyone has any idea then it will be a great help.
Thanks

**Comment by icyJoseph** [comment, marked as answer]
I think the issue is that when you have  `?amp`, it is mapped to a key;value pair, where key is `amp` and value is empty string, `""`.

Then a boolean cast is done on the value, like so, `Boolean(query.amp)`.

One thing you could try is to rewrite with middleware. I am not sure this works at all with `next.config.js`.

```ts
// _middleware.ts
import { NextResponse, type NextRequest } from "next/server";

export function middleware(req: NextRequest) {
  const amp = req.nextUrl.searchParams.get("amp");
  // strictly check that amp is empty string
  if (amp === "") {
    req.nextUrl.searchParams.set("amp", "1");
    return NextResponse.rewrite(req.nextUrl);
  }

  return NextResponse.next();
}
```

**Comment by yudiz-kuldip** [comment]
Hi @icyJoseph,
Thanks for the quick response.

I think **middleware** only works in the Vercel server.
Do you have any idea about this?

**Reply by icyJoseph** [reply]
On Vercel, it gets distributed to the edge, otherwise it should just run together with the server. Does it work if you run `npm run build && npm run start` locally? 

Just to make it clear, middleware works anywhere, because it just works with `next start`.  https://twitter.com/leeerob/status/1454247996737134592, I also had one of our apps in GCP, and now in AWS, and no special care has been given to middleware, other than writing it.

**Reply by yudiz-kuldip** [reply]
Hi @icyJoseph,
First of all Thanks for the help, You save my lots of time.
Sorry to disturb you again but I have one more question.

As per your answer, it's working properly.
But now I want to change the link tag URL as it is auto-generated by next itself.
Do you have any idea how can i change this URL?
Please refere the below image:
![image](https://user-images.githubusercontent.com/84446504/172124253-be30e23a-dd39-4db8-9b9b-322c6a00ed89.png)
